core/src/main/java/io/undertow/util/Cookies.java

```diff
@@ -211,51 +212,80 @@ public static Map<String, Cookie> parseRequestCookies(int maxCookies, boolean al
         return parseRequestCookies(maxCookies, allowEqualInValue, cookies, LegacyCookieSupport.COMMA_IS_SEPARATOR);
     }
 
+    @Deprecated(since="2.3.1", forRemoval=true)
     public static void parseRequestCookies(int maxCookies, boolean allowEqualInValue, List<String> cookies, Set<Cookie> parsedCookies) {
         parseRequestCookies(maxCookies, allowEqualInValue, cookies, parsedCookies, LegacyCookieSupport.COMMA_IS_SEPARATOR);
     }
 
+    public static void parseRequestCookies(int maxCookies, boolean allowEqualInValue, List<String> cookies, MultiValueHashListStorage<String, Cookie> parsedCookies) {
+        parseRequestCookies(maxCookies, allowEqualInValue, cookies, parsedCookies, LegacyCookieSupport.COMMA_IS_SEPARATOR);
+    }
+
     @Deprecated
     static Map<String, Cookie> parseRequestCookies(int maxCookies, boolean allowEqualInValue, List<String> cookies, boolean commaIsSeperator) {
         return parseRequestCookies(maxCookies, allowEqualInValue, cookies, commaIsSeperator, LegacyCookieSupport.ALLOW_HTTP_SEPARATORS_IN_V0);
     }
 
+    @Deprecated(since="2.3.1", forRemoval=true)
```

Comment:
Since this is public method please add javadoc that will contain at least this line and will guide developers in their migration path:

@deprecated use {@link #parseRequestCookies(int, boolean, List, MultiValueHashListStorage)} instead